File: leetcode8.cpp
class Solution {
public:
    int myAtoi(string str) {
        int sign = 1, i = 0 , n=str.length();
        long long result=0;
        while (i<n && str[i] == ' ') {i++;}
        if (i<n && str[i] == '-'){
            sign = -1;
            i++;
        }
        else if (i<n && str[i] == '+'){
            sign = 1;
            i++;
        }
        while (i<n && str[i] >= '0' && str[i] <= '9') {
            result  = 10 * result + (str[i++] - '0');
            if(result*sign >= INT_MAX) return INT_MAX;
            if(result*sign <= INT_MIN) return INT_MIN;
        }
        return result * sign;
    }
};

Runtime: 0 ms, faster than 100.00% of C++ online submissions for String to Integer (atoi).

File: leetcode8.py
class Solution:
    def myAtoi(self,str1):
        MAX_INT = 2147483647
        MIN_INT = -2147483648
        n,result,i,sign=len(str1),0,0,1
        while i<n and str1[i]==" ":
            i+=1
        if i<n and str1[i]=="-":
            sign=-1
            i+=1
        elif i<n and str1[i]=="+":
            sign=1
            i+=1
        while i<n and "0"<=str1[i]<="9":
            result=result*10+ord(str1[i])-ord("0")
            i+=1
        result=sign*result
        if result<MIN_INT: 
            return MIN_INT
        elif result>MAX_INT:
            return MAX_INT
        else:
            return result
        
python中没有MAX_INT MIN_INT这样的常量
